### PR TITLE
Add CODEOWNERS file

### PR DESCRIPTION
**Pull request by udim**
This file will enable auto-assignment of reviewers for Github PRs.

**Note:** The `@person1 @person2 @person3` placeholders will be removed in the final version of this file.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | --- | ---

### CODE REVIEW

**Comment by pabloem** [comment]
I've added myself for a few components. LMK if they're too many : ) - or if I may be appropriate owner for other ones too.

**Comment by jbonofre** [comment]
I added myself on some components.

**Comment by herohde** [comment]
Added myself to some components as well

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+/sdks/java/ @person1 @person2 @person3
+/sdks/java/core @lcwik @person2 @person3
+/sdks/java/extensions @person1 @person2 @person3
+/sdks/java/extensions/google-cloud-platform-core @lcwik @person2 @person3
+/sdks/java/extensions/jackson @person1 @person2 @person3
+/sdks/java/extensions/join-library @person1 @person2 @person3
+/sdks/java/extensions/protobuf @person1 @person2 @person3
+/sdks/java/extensions/sketching @person1 @person2 @person3
+/sdks/java/extensions/sorter @person1 @person2 @person3
+/sdks/java/extensions/sql @person1 @person2 @person3
```
Comment (akedin) [inline CODEOWNERS]: @akedin @apilloud @amaliujia @XuMingmin @xumingming

Comment (lukecwik) [reply CODEOWNERS]: Done

Comment (amaliujia) [reply CODEOWNERS]: Thanks!

**Comment by timrobertson100** [comment]
I'm not a committer but have been reviewing and contributing.  Can you please add me @timrobertson100 to the following if appropriate?

```
/website                                  
/website/src                              

/sdks/java/io
/sdks/java/io/common                      
/sdks/java/io/file-based-io-tests         
/sdks/java/io/elasticsearch-tests         
/sdks/java/io/hadoop-input-format         
/sdks/java/io/solr                        
/sdks/java/io/hcatalog                    
/sdks/java/io/elasticsearch               
/sdks/java/io/jdbc                        
/sdks/java/io/hadoop-file-system          
/sdks/java/io/hbase                       
/sdks/java/io/hadoop-common               
```

**Comment by udim** [comment]
@timrobertson100 done!

**Comment by timrobertson100** [comment]
Thanks @udim 

I've asked GitHub helpdesk if those with non write access can be added, and if it round robins when there are multiple people. I'll post the reply to your thread on the @dev mail list.

**Pull request merged**